keps/sig-release/1731-publishing-packages/README.md

```diff
@@ -149,51 +183,236 @@ Scenario: [...]
+
+- Ubuntu 20.04 (`aarch64`, `armv7l`, `ppc64le`, `s390x`, `x86_64`)
+- CentOS Stream 8 (`aarch64`, `ppc64le`, `x86_64`)
+  - CentOS Stream 8 doesn't support `armv7l` and `s390x` on OBS, hence, we use OpenSUSE Factory which does support
+    those architectures
+- OpenSUSE Factory (`armv7l`, `s390x`)
+
+Those operating systems are used as **builders**. The resulting packages are supposed to work on any Debian-based
+and RPM-based distribution, respectively. Those distributions were chosen based on recommendations from the OBS team
+to provide the best compatibility with all all Debian-based and RPM-based distributions.
+
+The following packages will be published for all operating system and architectures listed earlier. For simplicity,
+we'll refer to those as packages as the **core packages**:
+
+- cri-tools
+- kubeadm
+- kubectl
+- kubelet
+- kubernetes-cni
+
+#### Repository Layout
+
+We'll use this layout when creating repositories for the **core packages**:
 
-- Package managers, like `apt` (consumes deb packages) and `yum` (consumes rpm packages).
-- **`${k8s_release}`**: the version of kubernetes `<major>.<minor>`
-  (e.g. `1.12`, `1.13`, `1.14`, ...)
 - **`${channel}`**: can be `stable`, `dev`, `nightly`
   - `stable`: all official releases for `${k8s_release}`
-    (e.g.: `1.13.0`, `1.13.1`, `1.13.2`, ...)
-  - `dev`: all development releases for all minor releases in this `${k8s_release}`, including `alpha`s, `beta`s and `rc`s
-    (e.g.: `1.13.0-rc.2`, `1.13.2-beta.0`, `1.13.1-alpha.3`, ...)
-  - `nightly`: any package cut automatically on a daily basis (optionally)
+    (e.g.: `1.26.0`, `1.26.1`, `1.26.2`, ...)
+  - `dev`: all development releases for all minor releases in this `${k8s_release}`,
+    including `alpha`s, `beta`s and `rc`s (e.g.: `1.26.0-rc.2`, `1.26.2-beta.0`, `1.26.1-alpha.3`, ...)
+  - `nightly`: any package cut automatically from the `master` branch on a daily basis (optionally)
+    - Nightly packages are currently out of scope and might be handled via a different KEP
+- **`${k8s_release}`**: the version of Kubernetes `<major>.<minor>`
+  (e.g. `1.12`, `1.13`, `1.14`, ...)
 
-This means, that End Users can configure their systems’ package managers to use
-those different `${channel}`s of a kubernetes `${k8s_release}` for their
-corresponding package manager.
+#### Projects and Packages in OBS
+
+Speaking of OBS, **Packages** are located in **Projects**. We'll use two different types of projects:
+
+- Building/Staging project - we'll build packages in a project of this type
+- Publishing/Maintenance project - we'll publish packages from a project of this type
+
+The reason for using two different types of projects is that OBS doesn't keep previously published packages.
+For example, if we publish v1.26.0, and then v1.26.1, after v1.26.1 is published, the v1.26.0 package is removed.
+To be able to keep all packages/versions (e.g. v1.26.0 and v1.26.1), we need to publish packages from the
+maintenance project.
```

Comment:
> OBS doesn't keep previously published packages

Is there any roadmap item to fix this or add a flag for builds? Maybe it's something we could contribute to OBS.

Reply:
Since OBS auto-rebuilds packages when underlying dependencies change, the capability has been intentionally left out to conserve space.

Reply:
Agreed, this is the recommended approach. maybe the wording should be enhanced to be less misleading here. Suggest something like

`
The reason for using two different types of projects is that in the default OBS projects, which are of build-type, only one set of binaries is kept at any point in time and used to provide build dependencies for all the artifacts to build. By releasing built binaries into  a maintenance-type project, they can be snapshot-ed and stored together. The published repository includes all snapshotted binaries.
`

Reply:
I like this wording, thank you! I'll add it to the KEP.